datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -850,6 +864,79 @@ pub async fn from_substrait_rel(
     }
 }
 
+/// Ensures that the given Substrait schema is compatible with the schema as given by DataFusion
+///
+/// This means:
+/// 1. All fields present in the Substrait schema are present in the DataFusion schema. The
+///    DataFusion schema may have MORE fields, but not the other way around.
+/// 2. All fields are compatible. See [`ensure_field_compatability`] for details
+///
+/// This function returns a DataFrame with fields adjusted if necessary in the event that the
+/// Substrait schema is a subset of the DataFusion schema.
+fn ensure_schema_compatability(
+    table: DataFrame,
+    substrait_schema: DFSchema,
+) -> Result<DataFrame> {
+    // For now strip the qualifiers from the DataFusion and Substrait schemas to make comparisons
+    // easier as there are issues around qualifier case.
+    // TODO: ensure that qualifiers are the same
```

Comment:
FWIW I don't think it's super necessary/useful to ensure the qualifiers even in future, presumably if DF doesn't have a table with a matching qualifier (names) the query will fail 🤷

Reply:
I see what you're saying. The table lookup acts like an implicit check on the qualifier. I'll remove this.